refinery/ui/source/js/dashboard/services/data-set-card-factory.js

```diff
@@ -23,10 +23,12 @@
   ) {
     var dataSets = [];
     var currentParams = {};
+    var dataSetStats = { totalCount: 0 };
```

Comment:
For just one value, wrapping it in a dict seems unnecessary?

Reply:
This is something good to point out, since you may encounter it in the future. Avoid binding to primitives. I tried to find an explanation online.
 https://stackoverflow.com/questions/27518408/angualrjs-why-can-only-bind-to-reference-values-but-primitive-values-of-service

This is also a good link discussing javascript and angularjs's scoping.
https://github.com/angular/angular.js/wiki/Understanding-Scopes

Reply:
@mccalluc

Reply:
Refactored section
https://github.com/refinery-platform/refinery-platform/pull/2926